### PR TITLE
PWX-31985: Improves FA snap/clone test logging

### PR DESCRIPTION
**Pull request by Pure-AdamuKaapan**
**What this PR does / why we need it**:
There seems to be some size discrepancies between the restored PVC being created and the original PVC, but it seems like it may be on the test side. I've also been meaning to improve some of the logging here, so threw in some other fixes as well:
* Waits for PVCs to bind before waiting for apps to bind, reducing confusion over which part failed
* Prints size of original PVC before restoring
* Various typo fixes

Will test before merging

### CODE REVIEW

**Review by jainsmit — APPROVED** [review]
LGTM !!

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -6320,18 +6320,19 @@ func (k *K8s) CSISnapshotTest(ctx *scheduler.Context, request scheduler.CSISnaps
...
 
 	if err != nil {
 		log.Errorf("Failed to retrieve PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
 		return err
 	}
 	originalStorageClass, err := k8sCore.GetStorageClassForPVC(pvcObj)
 	if err != nil {
-		log.Errorf("Failed to retrieve SC for PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
+		log.Errorf("Failed to retrieve storage class for PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
 		return err
```
[the remark this answers is not in the document]
Comment (Pure-AdamuKaapan) [reply drivers/scheduler/k8s/k8s.go:6332]: Sounds good! Will go through and do that in a bunch of places

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -7014,8 +7005,34 @@ func (k *K8s) waitForPodToBeReady(podname string, namespace string) error {
...
 
-// waitForRestoredPVCsToBound retries and wait up to 30 minutes for all PVCs to bound
-func (k *K8s) waitForRestoredPVCsToBound(pvcNamePrefix string, namespace string) error {
+// waitForRestoredPVCsToBound retries and waits up to 30 minutes for a single PVC to be bound
+func (k *K8s) waitForSinglePVCToBound(pvcName, namespace string) error {
+	var pvc *v1.PersistentVolumeClaim
+	var err error
+	kubeClient, err := k.getKubeClient("")
+	if err != nil {
+		log.Error("Failed to get Kube client")
```
Comment (nikolaypopov) [inline drivers/scheduler/k8s/k8s.go:7014]: Just return this in the error, if needed

**Review by nikolaypopov — APPROVED** [review]
Minor nit, otherwise gltm

**Pull request merged**